diff --git a/build/pgo/profileserver.py b/build/pgo/profileserver.py
--- a/build/pgo/profileserver.py
+++ b/build/pgo/profileserver.py
@@ -130,11 +130,11 @@
 
         # Ensure different pids write to different files
         env["LLVM_PROFILE_FILE"] = "default_%p_random_%m.profraw"
 
         # Write to an output file if we're running in automation
-        process_args = {}
+        process_args = {'universal_newlines': True}
         if 'UPLOAD_PATH' in env:
             process_args['logfile'] = os.path.join(env['UPLOAD_PATH'], 'profile-run-1.log')
 
         # Run Firefox a first time to initialize its profile
         runner = FirefoxRunner(profile=profile,

