--- a/third_party///libdrm/BUILD.gn	2019-05-17 17:48:02.036000000 +0300
+++ b/third_party///libdrm/BUILD.gn	2019-05-17 18:56:17.612000000 +0300
@@ -1,22 +1,75 @@
-# Copyright 2017 The Chromium Authors. All rights reserved.
+# Copyright 2016 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
+assert(is_linux)
 
-import("//build/config/linux/pkg_config.gni")
-import("//build/shim_headers.gni")
+config("libdrm_config") {
+  # TODO(thomasanderson): Remove this hack once
+  # https://patchwork.kernel.org/patch/10545295/ lands.
+  defines = [ "typeof(x)=__typeof__(x)" ]
 
-pkg_config("system_libdrm") {
-  packages = [ "libdrm" ]
+  include_dirs = [
+    "src",
+    "binclude",
+    "binclude/drm",
+  ]
+
+  # libdrm uses macros defined by <sys/types.h> which are being moved to
+  # <sys/sysmacros.h>.  GLIBC headers give a pragma warning in this case.
+  # Suppress this warning for now.  This may be removed once
+  # https://patchwork.kernel.org/patch/9628231/ lands.
+  cflags = [ "-Wno-#pragma-messages" ]
+  if (is_clang) {
+    cflags += [ "-Wno-enum-conversion" ]
+  }
 }
 
-shim_headers("libdrm_shim") {
-  root_path = "binclude"
-  headers = [ "drm.h" ]
+static_library("libdrm") {
+  sources = [
+    "bxf86drm.c",
+    "bxf86drmHash.c",
+    "bxf86drmMode.c",
+    "bxf86drmRandom.c",
+  ]
+
+  include_dirs = [
+    "src",
+    "binclude",
+  ]
+  configs -= [ "//build/config/compiler:chromium_code" ]
+  configs += [ "//build/config/compiler:no_chromium_code" ]
+  cflags = [
+    # xf86drm.c uses readdir_r, which has been deprecated as of
+    # glibc-2.24.  This causes a build error when using the Debian
+    # Stretch sysroot.
+    "-Wno-deprecated-declarations",
+  ]
+
+  public_configs = [ ":libdrm_config" ]
 }
 
-source_set("libdrm") {
+executable("modetest") {
+  sources = [
+    "btests/modetest/buffers.c",
+    "btests/modetest/buffers.h",
+    "btests/modetest/cursor.c",
+    "btests/modetest/cursor.h",
+    "btests/modetest/modetest.c",
+    "btests/util/common.h",
+    "btests/util/format.c",
+    "btests/util/format.h",
+    "btests/util/kms.c",
+    "btests/util/kms.h",
+    "btests/util/pattern.c",
+    "btests/util/pattern.h",
+  ]
+
+  include_dirs = [
+    "btests",
+    "btests/modetest",
+  ]
+
   deps = [
-    ":libdrm_shim",
+    ":libdrm",
   ]
-  public_configs = [ ":system_libdrm" ]
 }
--- a/third_party///libdrm/OWNERS	1970-01-01 03:00:00.000000000 +0300
+++ b/third_party///libdrm/OWNERS	2019-05-17 18:56:17.612000000 +0300
@@ -0,0 +1,6 @@
+dcastagna@chromium.org
+hoegsberg@chromium.org
+reveman@chromium.org
+spang@chromium.org
+servolk@chromium.org
+dnicoara@chromium.org
--- a/third_party///libdrm/README.chromium	1970-01-01 03:00:00.000000000 +0300
+++ b/third_party///libdrm/README.chromium	2019-05-17 18:56:17.612000000 +0300
@@ -0,0 +1,13 @@
+Name: libdrm
+Short Name: libdrm
+URL: https://chromium.googlesource.com/chromiumos/third_party/libdrm
+Version: 2.4.85
+License: MIT, GPL
+License File: NOT_SHIPPED
+Security Critical: yes
+
+Description:
+Userspace interface to kernel DRM services.
+
+Local Modifications:
+None
