diff --git a/build/config/ui.gni b/build/config/ui.gni
index 547b42f..966b00c 100644
--- a/build/config/ui.gni
+++ b/build/config/ui.gni
@@ -51,8 +51,10 @@ if (use_ozone) {
 # Make sure glib is not used if building for ChromeOS/Chromecast
 assert(!use_glib || (is_linux && !is_chromeos && !is_chromecast))
 
-# Whether to use atk, the Accessibility ToolKit library
-use_atk = is_desktop_linux && !is_chromecast && use_glib
+declare_args() {
+  # Whether to use atk, the Accessibility ToolKit library
+  use_atk = is_desktop_linux && !is_chromecast && use_glib
+}
 # =============================================
 #   PLEASE DO NOT ADD MORE FLAGS TO THIS FILE
 # =============================================
