--- chromium-88.0.4324.41/build/linux/unbundle/libvpx.gn.system-libvpx	2020-12-10 13:02:40.000000000 +0100
+++ chromium-88.0.4324.41/build/linux/unbundle/libvpx.gn	2020-12-12 18:44:29.054701046 +0100
@@ -26,7 +26,15 @@ shim_headers("libvpx_shim") {
   ]
 }
 
+config("libvpx_config") {
+}
+
 source_set("libvpx") {
   deps = [ ":libvpx_shim" ]
   public_configs = [ ":system_libvpx" ]
 }
+
+source_set("libvp9rc") {
+  deps = [ ":libvpx_shim" ]
+  public_configs = [ ":system_libvpx" ]
+}
