--- chromium-73.0.3683.75/components/download/internal/common/download_item_impl.cc.gcc8-5	2019-03-17 16:29:19.104259012 +0100
+++ chromium-73.0.3683.75/components/download/internal/common/download_item_impl.cc	2019-03-17 20:25:57.404313744 +0100
@@ -646,7 +646,7 @@ void DownloadItemImpl::UpdateResumptionI
     bytes_wasted_ = 0;
   }
 
-  auto_resume_count_ = user_resume ? 0 : ++auto_resume_count_;
+  auto_resume_count_ = user_resume ? 0 : auto_resume_count_+1;
   download_schedule_ = absl::nullopt;
   RecordDownloadLaterEvent(DownloadLaterEvent::kScheduleRemoved);
 }
--- chromium-74.0.3729.40/third_party/blink/renderer/modules/animationworklet/animation_worklet_proxy_client.cc.gcc8	2019-03-27 19:50:26.000000000 +0100
+++ chromium-74.0.3729.40/third_party/blink/renderer/modules/animationworklet/animation_worklet_proxy_client.cc	2019-03-31 22:58:52.566976070 +0200
@@ -185,7 +185,7 @@ AnimationWorkletProxyClient::SelectGloba
   if (--next_global_scope_switch_countdown_ < 0) {
     int last_global_scope_index = current_global_scope_index_;
     current_global_scope_index_ =
-        (++current_global_scope_index_ % global_scopes_.size());
+        ((current_global_scope_index_ + 1) % global_scopes_.size());
     global_scopes_[last_global_scope_index]->MigrateAnimatorsTo(
         global_scopes_[current_global_scope_index_]);
     // Introduce an element of randomness in the switching interval to make
--- chromium-75.0.3766.2/chrome/browser/notifications/scheduler/internal/impression_history_tracker.cc.gcc8	2019-04-16 03:13:59.000000000 +0200
+++ chromium-75.0.3766.2/chrome/browser/notifications/scheduler/internal/impression_history_tracker.cc	2019-04-21 10:17:31.053438403 +0200
@@ -414,7 +414,7 @@ void ImpressionHistoryTrackerImpl::Apply
 
   // Increase |current_max_daily_show| by 1.
   client_state->current_max_daily_show =
-      base::clamp(++client_state->current_max_daily_show, 0,
+      base::clamp(client_state->current_max_daily_show + 1, 0,
                   config_.max_daily_shown_per_type);
 }
 
