--- chromium-77.0.3865.10/media/base/mime_util_internal.cc.android	2019-08-11 20:20:57.124817365 +0200
+++ chromium-77.0.3865.10/media/base/mime_util_internal.cc	2019-08-11 20:40:50.617866174 +0200
@@ -629,6 +629,7 @@ SupportsType MimeUtil::IsSupportedMediaF
   return AreSupportedCodecs(parsed_results, mime_type_lower_case, is_encrypted);
 }
 
+#if defined(OS_ANDROID)
 // static
 bool MimeUtil::IsCodecSupportedOnAndroid(
     Codec codec,
@@ -761,6 +762,7 @@ bool MimeUtil::IsCodecSupportedOnAndroid
 
   return false;
 }
+#endif
 
 bool MimeUtil::ParseCodecStrings(
     const std::string& mime_type_lower_case,
