--- chromium-94.0.4606.31/components/translate/core/browser/translate_prefs.cc.startup-crash	2021-09-07 23:16:49.604955369 +0200
+++ chromium-94.0.4606.31/components/translate/core/browser/translate_prefs.cc	2021-09-07 23:17:23.215303436 +0200
@@ -574,7 +574,7 @@ std::vector<std::string> TranslatePrefs:
   for (auto entry : dict->DictItems()) {
     absl::optional<base::Time> time = base::ValueToTime(entry.second);
     if (!time) {
-      NOTREACHED();
+//      NOTREACHED();
       continue;
     }
     if (begin <= *time && *time < end)
