diff -up a/chrome/browser/profiling_host/BUILD.gn b/chrome/browser/profiling_host/BUILD.gn
--- a/chrome/browser/profiling_host/BUILD.gn	2021-02-12 16:57:09.254914601 +0100
+++ b/chrome/browser/profiling_host/BUILD.gn	2021-02-12 16:57:18.958931656 +0100
@@ -22,7 +22,6 @@ static_library("profiling_host") {
     "//components/version_info",
     "//content/public/browser",
     "//content/public/common",
-    "//third_party/zlib:zlib",
 
     # Added to support the dependency on //chrome/browser/metrics
     # TODO(darin): We should instead have a way to express that dependency here.
