--- chromium-91.0.4472.77/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc.no-external-components	2021-05-25 00:05:34.000000000 +0200
+++ chromium-91.0.4472.77/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc	2021-05-27 10:39:16.993258562 +0200
@@ -28,22 +28,7 @@ namespace extensions {
 
 bool IsComponentExtensionAllowlisted(const std::string& extension_id) {
   const char* const kAllowed[] = {
-    extension_misc::kInAppPaymentsSupportAppId,
     extension_misc::kPdfExtensionId,
-#if defined(OS_CHROMEOS)
-    extension_misc::kAssessmentAssistantExtensionId,
-#endif
-#if BUILDFLAG(IS_CHROMEOS_ASH)
-    extension_misc::kAccessibilityCommonExtensionId,
-    extension_misc::kCameraAppId,
-    extension_misc::kChromeVoxExtensionId,
-    extension_misc::kEnhancedNetworkTtsExtensionId,
-    extension_misc::kEspeakSpeechSynthesisExtensionId,
-    extension_misc::kGoogleSpeechSynthesisExtensionId,
-    extension_misc::kGuestModeTestExtensionId,
-    extension_misc::kSelectToSpeakExtensionId,
-    extension_misc::kSwitchAccessExtensionId,
-#endif
   };
 
   for (size_t i = 0; i < base::size(kAllowed); ++i) {
