--- chromium-58.0.3029.81/third_party/node/node.py.system-nodejs	2017-04-19 21:06:52.000000000 +0200
+++ chromium-58.0.3029.81/third_party/node/node.py	2017-04-24 00:52:35.271085493 +0200
@@ -11,17 +11,7 @@ import os
 
 
 def GetBinaryPath():
-  # TODO: Node 16.0 will likely ship with an official universal node binary
-  # on macOS. Once node 16.0 is released, remove this special case here
-  # and use node-darwin-universal in the dict in the main return statement.
-  if platform.system() == 'Darwin' and platform.machine() == 'arm64':
-      return os.path.join(os_path.join(os_path.dirname(__file__), 'mac',
-                          'node-darwin-arm64', 'bin', 'node'))
-  return os_path.join(os_path.dirname(__file__), *{
-    'Darwin': ('mac', 'node-darwin-x64', 'bin', 'node'),
-    'Linux': ('linux', 'node-linux-x64', 'bin', 'node'),
-    'Windows': ('win', 'node.exe'),
-  }[platform.system()])
+  return '/usr/bin/node'
 
 
 def RunNode(cmd_parts, stdout=None):
