--- a/python/mozbuild/mozbuild/action/langpack_manifest.py
+++ b/python/mozbuild/mozbuild/action/langpack_manifest.py
@@ -42,8 +42,8 @@ pushlog_api_url = "{0}/json-rev/{1}"
 
 def get_build_date():
     """Return the current date or SOURCE_DATE_EPOCH, if set."""
-    return datetime.datetime.utcfromtimestamp(
-        int(os.environ.get("SOURCE_DATE_EPOCH", time.time()))
+    return datetime.datetime.fromtimestamp(
+        int(os.environ.get("SOURCE_DATE_EPOCH", time.time())), datetime.UTC
     )
 
 
@@ -92,7 +92,7 @@ def get_dt_from_hg(path):
         msg = f"{str(exc)}\ndata is: {json.dumps(data, indent=2, sort_keys=True)}"
         raise KeyError(msg)
 
-    return datetime.datetime.utcfromtimestamp(date)
+    return datetime.datetime.fromtimestamp(date, datetime.UTC)
 
 
 ###
