diff --git a/chrome/browser/extensions/chrome_component_extension_resource_manager.cc b/chrome/browser/extensions/chrome_component_extension_resource_manager.cc
index cefe299..ecde539 100644
--- a/chrome/browser/extensions/chrome_component_extension_resource_manager.cc
+++ b/chrome/browser/extensions/chrome_component_extension_resource_manager.cc
@@ -20,12 +20,12 @@
 #include "chrome/browser/chromeos/file_manager/file_manager_string_util.h"
 #include "chrome/grit/component_extension_resources.h"
 #include "extensions/common/constants.h"
-#include "third_party/ink/grit/ink_resources.h"
+// #include "third_party/ink/grit/ink_resources.h"
 #include "ui/base/resource/resource_bundle.h"
 #include "ui/file_manager/file_manager_resource_util.h"
 #include "ui/file_manager/grit/file_manager_resources.h"
 #endif
-
+  
 #if defined(KIOSK_NEXT) && defined(GOOGLE_CHROME_BUILD)
 #include "chrome/grit/kiosk_next_internal_resources_map.h"
 #endif
@@ -51,12 +51,12 @@ ChromeComponentExtensionResourceManager() {
 #if defined(OS_CHROMEOS)
     {"chrome_app/chrome_app_icon_32.png", IDR_CHROME_APP_ICON_32},
     {"chrome_app/chrome_app_icon_192.png", IDR_CHROME_APP_ICON_192},
-    {"pdf/ink/ink_lib_binary.js", IDR_INK_LIB_BINARY_JS},
-    {"pdf/ink/pthread-main.js", IDR_INK_PTHREAD_MAIN_JS},
-    {"pdf/ink/glcore_base.js.mem", IDR_INK_GLCORE_BASE_JS_MEM},
-    {"pdf/ink/glcore_base.wasm", IDR_INK_GLCORE_BASE_WASM},
-    {"pdf/ink/glcore_wasm_bootstrap_compiled.js",
-     IDR_INK_GLCORE_WASM_BOOTSTRAP_COMPILED_JS},
+//    {"pdf/ink/ink_lib_binary.js", IDR_INK_LIB_BINARY_JS},
+//    {"pdf/ink/pthread-main.js", IDR_INK_PTHREAD_MAIN_JS},
+//    {"pdf/ink/glcore_base.js.mem", IDR_INK_GLCORE_BASE_JS_MEM},
+//    {"pdf/ink/glcore_base.wasm", IDR_INK_GLCORE_BASE_WASM},
+//    {"pdf/ink/glcore_wasm_bootstrap_compiled.js",
+//     IDR_INK_GLCORE_WASM_BOOTSTRAP_COMPILED_JS},
 #endif
   };
 
