diff --git a/clients/meson.build b/clients/meson.build
index 47e9e8ce..82e36d24 100644
--- a/clients/meson.build
+++ b/clients/meson.build
@@ -270,7 +270,11 @@ demo_clients = [
 		'basename': 'subsurfaces',
 		'deps': [ 'egl', 'glesv2', 'wayland-egl' ]
 	},
-	{ 'basename': 'transformed' },
+	{ 'basename': 'transformed' },   
+	{
+		'basename': 'gears',
+		'deps': [ 'egl', 'gl', 'cairo', 'wayland-egl' ]
+	}, 
 ]
 
 if get_option('demo-clients')
