diff --git a/include/linux/if_link.h b/include/linux/if_link.h
index e3e1008..5195ed9 100644
--- a/include/linux/if_link.h
+++ b/include/linux/if_link.h
@@ -1258,25 +1258,4 @@ struct ifla_rmnet_flags {
 	__u32	mask;
 };
 
-/* rmnet section */
-
-#define RMNET_FLAGS_INGRESS_DEAGGREGATION         (1U << 0)
-#define RMNET_FLAGS_INGRESS_MAP_COMMANDS          (1U << 1)
-#define RMNET_FLAGS_INGRESS_MAP_CKSUMV4           (1U << 2)
-#define RMNET_FLAGS_EGRESS_MAP_CKSUMV4            (1U << 3)
-
-enum {
-	IFLA_RMNET_UNSPEC,
-	IFLA_RMNET_MUX_ID,
-	IFLA_RMNET_FLAGS,
-	__IFLA_RMNET_MAX,
-};
-
-#define IFLA_RMNET_MAX	(__IFLA_RMNET_MAX - 1)
-
-struct ifla_rmnet_flags {
-	__u32	flags;
-	__u32	mask;
-};
-
 #endif /* _LINUX_IF_LINK_H */
