--- a/conf/gkbuild.cmake
+++ b/conf/gkbuild.cmake
@@ -30,13 +30,13 @@ if(CMAKE_COMPILER_IS_GNUCC)
 # GCC opts.
   set(GK_COPTIONS "${GK_COPTIONS} -std=c99 -fno-strict-aliasing")
 if(VALGRIND)
-  set(GK_COPTIONS "${GK_COPTIONS} -march=x86-64 -mtune=generic")
+  set(GK_COPTIONS "${GK_COPTIONS}")
 else()
 # -march=native is not a valid flag on PPC:
 if(CMAKE_SYSTEM_PROCESSOR MATCHES "power|ppc|powerpc|ppc64|powerpc64" OR (APPLE AND CMAKE_OSX_ARCHITECTURES MATCHES "ppc|ppc64"))
-  set(GK_COPTIONS "${GK_COPTIONS} -mtune=native")
+  set(GK_COPTIONS "${GK_COPTIONS}")
 else()
-  set(GK_COPTIONS "${GK_COPTIONS} -march=native")
+  set(GK_COPTIONS "${GK_COPTIONS}")
 endif()
 endif(VALGRIND)
 # GCC warnings.
