diff --git a/build/config/ui.gni b/build/config/ui.gni
index 022f008..bb1d3f0 100644
--- a/build/config/ui.gni
+++ b/build/config/ui.gni
@@ -51,7 +51,9 @@ assert(!use_glib || ((is_linux || is_chromeos_lacros) && !is_chromecast))
 
 # TODO(crbug.com/1171629): Remove is_chromeos_lacros.
 # Whether to use atk, the Accessibility ToolKit library
-use_atk = (is_linux || is_chromeos_lacros) && !is_chromecast && use_glib
+declare_args() {  
+    use_atk = (is_linux || is_chromeos_lacros) && !is_chromecast && use_glib
+}
 
 # Whether using Xvfb to provide a display server for a test might be
 # necessary.
