diff --git a/src/vabackend.c b/src/vabackend.c
index 5353d75..34de561 100644
--- a/src/vabackend.c
+++ b/src/vabackend.c
@@ -2186,11 +2186,11 @@ __attribute__((visibility("default")))
 VAStatus __vaDriverInit_1_0(VADriverContextP ctx) {
     LOG("Initialising NVIDIA VA-API Driver: %lX", ctx->display_type);
 
-    //drm_state can be passed in with any display type, including X11. But if it's X11, we don't
-    //want to use the fd as it'll likely be an Intel GPU, as NVIDIA doesn't support DRI3 at the moment
+    //drm_state can be passed in with any display type, including X11. But if it's X11 or wayland, we 
+    //don't want to use the fd as it'll likely be an Intel GPU, as NVIDIA doesn't support DRI3 at the
+    //moment
     bool isDrm = ctx->drm_state != NULL && ((struct drm_state*) ctx->drm_state)->fd > 0 &&
-                 (((ctx->display_type & VA_DISPLAY_MAJOR_MASK) == VA_DISPLAY_DRM) ||
-                  ((ctx->display_type & VA_DISPLAY_MAJOR_MASK) == VA_DISPLAY_WAYLAND));
+                 ((ctx->display_type & VA_DISPLAY_MAJOR_MASK) == VA_DISPLAY_DRM);   
 
     pthread_mutex_lock(&concurrency_mutex);
     LOG("Now have %d (%d max) instances", instances, max_instances);
