--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -1112,7 +1112,7 @@ def compiler_wrapper(wrapper, ccache):
 
 @dependable
 def wasm():
-    return split_triplet("wasm32-wasi", allow_wasi=True)
+    return split_triplet("wasm32-wasip1", allow_wasi=True)
 
 
 @template
--- a/gfx/harfbuzz/src/wasm/sample/c/Makefile
+++ b/gfx/harfbuzz/src/wasm/sample/c/Makefile
@@ -4,7 +4,7 @@ all: test-fallback.wasm.ttf test-ot.wasm
 
 %.wasm: %.cc ../../../hb-wasm-api.h
 	clang \
-		--target=wasm32-unknown-wasi \
+		--target=wasm32-unknown-wasip1 \
 		-Wl,--no-entry \
 		-fvisibility=hidden \
 		-Wl,--allow-undefined \
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -2939,7 +2939,7 @@ with only_when(requires_wasm_sandboxing
     def wasi_sysroot_flags(wasi_sysroot):
         if wasi_sysroot:
             log.info("Using wasi sysroot in %s", wasi_sysroot)
-            return ["--sysroot=%s" % wasi_sysroot, "-nodefaultlibs",  "-lc", "-lwasi-emulated-process-clocks", "-lc++", "-lc++abi", "/var/tmp/portage/www-client/zen-browser-1.19.3_beta/work/wasi-sdk-32.0-x86_64-linux/lib/clang/22/lib/wasm32-unknown-wasi/libclang_rt.builtins.a"]
+            return ["--sysroot=%s" % wasi_sysroot, "-nodefaultlibs",  "-lc", "-lwasi-emulated-process-clocks", "-lc++", "-lc++abi", "/var/tmp/portage/www-client/zen-browser-1.19.3_beta/work/wasi-sdk-32.0-x86_64-linux/lib/clang/22/lib/wasm32-unknown-wasip1/libclang_rt.builtins.a"]
         return []
 
     set_config("WASI_SYSROOT", wasi_sysroot)
