--- a/toolkit/components/moz.build
+++ b/toolkit/components/moz.build
@@ -141,7 +141,10 @@ if CONFIG["MOZ_WIDGET_TOOLKIT"] not in (
     DIRS += ["aboutwebauthn"]
 
 if CONFIG["MOZ_BUILD_APP"] == "browser":
-    DIRS += ["messaging-system", "normandy"]
+    DIRS += ["messaging-system"]
+
+if CONFIG["MOZ_NORMANDY"] == "1":
+    DIRS += ["normandy"]
 
 if CONFIG["MOZ_BUILD_APP"] != "comm/mail":
     DIRS += ["ipprotection"]
