diff --git a/chrome/browser/extensions/BUILD.gn b/chrome/browser/extensions/BUILD.gn
index 5aea558..9d72d5f 100644
--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -257,8 +257,8 @@ jumbo_static_library("extensions") {
     "api/module/module.h",
     "api/networking_private/networking_private_ui_delegate_chromeos.cc",
     "api/networking_private/networking_private_ui_delegate_chromeos.h",
-    "api/networking_private/networking_private_ui_delegate_factory_impl.cc",
-    "api/networking_private/networking_private_ui_delegate_factory_impl.h",
+#    "api/networking_private/networking_private_ui_delegate_factory_impl.cc",
+#    "api/networking_private/networking_private_ui_delegate_factory_impl.h",
     "api/notifications/extension_notification_display_helper.cc",
     "api/notifications/extension_notification_display_helper.h",
     "api/notifications/extension_notification_display_helper_factory.cc",
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -1203,8 +1203,8 @@ jumbo_split_static_library("ui") {
       "webui/settings/appearance_handler.h",
       "webui/settings/browser_lifetime_handler.cc",
       "webui/settings/browser_lifetime_handler.h",
-      "webui/settings/change_password_handler.cc",
-      "webui/settings/change_password_handler.h",
+#      "webui/settings/change_password_handler.cc",
+#      "webui/settings/change_password_handler.h",
       "webui/settings/custom_home_pages_table_model.cc",
       "webui/settings/custom_home_pages_table_model.h",
       "webui/settings/downloads_handler.cc",
