--- chromium-72.0.3626.81/tools/gn/build/gen.py.gn-bootstrap-2	2019-02-02 13:59:08.890057595 +0100
+++ chromium-72.0.3626.81/tools/gn/build/gen.py	2019-02-02 14:11:35.667096734 +0100
@@ -308,7 +308,7 @@ def WriteGNNinja(path, platform, host, o
     ar = os.environ.get('AR', 'ar')
   else:
     cxx = os.environ.get('CXX', 'clang++')
-    ld = cxx
+    ld = os.environ.get('LD', 'clang++')
     ar = os.environ.get('AR', 'ar')
 
   cflags = os.environ.get('CFLAGS', '').split()
