--- chromium-72.0.3626.96/third_party/blink/renderer/core/fetch/multipart_parser.cc.i586	2019-02-06 23:06:47.000000000 +0100
+++ chromium-72.0.3626.96/third_party/blink/renderer/core/fetch/multipart_parser.cc	2019-02-18 08:53:30.267915984 +0100
@@ -299,7 +299,7 @@ bool MultipartParser::ParseHeaderFields(
                                         HTTPHeaderMap* header_fields) {
   // Combine the current bytes with buffered header bytes if needed.
   const char* header_bytes = *bytes_pointer;
-  if ((bytes_end - *bytes_pointer) > std::numeric_limits<wtf_size_t>::max())
+  if (uintptr_t(bytes_end - *bytes_pointer) > std::numeric_limits<wtf_size_t>::max())
     return false;
 
   wtf_size_t header_size = static_cast<wtf_size_t>(bytes_end - *bytes_pointer);
