diff --git a/media/gpu/v4l2/BUILD.gn b/media/gpu/v4l2/BUILD.gn
index 755f8bc..617626a 100644
--- a/media/gpu/v4l2/BUILD.gn
+++ b/media/gpu/v4l2/BUILD.gn
@@ -39,8 +39,6 @@ source_set("v4l2") {
     "v4l2_framerate_control.h",
     "v4l2_h264_accelerator.cc",
     "v4l2_h264_accelerator.h",
-    "v4l2_h264_accelerator_legacy.cc",
-    "v4l2_h264_accelerator_legacy.h",
     "v4l2_image_processor_backend.cc",
     "v4l2_image_processor_backend.h",
     "v4l2_slice_video_decode_accelerator.cc",
@@ -63,14 +61,8 @@ source_set("v4l2") {
     "v4l2_video_decoder_backend_stateless.h",
     "v4l2_video_encode_accelerator.cc", 
     "v4l2_video_encode_accelerator.h",
-    "v4l2_vp8_accelerator.cc",
-    "v4l2_vp8_accelerator.h",
-    "v4l2_vp8_accelerator_legacy.cc",
-    "v4l2_vp8_accelerator_legacy.h",
     "v4l2_vp9_accelerator_chromium.cc",
     "v4l2_vp9_accelerator_chromium.h",
-    "v4l2_vp9_accelerator_legacy.cc",
-    "v4l2_vp9_accelerator_legacy.h",
   ]
 
   libs = [
--
