diff --git a/third_party/dawn/third_party/tint/src/sem/block_statement.h b/third_party/dawn/third_party/tint/src/sem/block_statement.h
index f8c8113..d6dc1f7 100644
--- a/third_party/dawn/third_party/tint/src/sem/block_statement.h
+++ b/third_party/dawn/third_party/tint/src/sem/block_statement.h
@@ -16,6 +16,7 @@
 #define SRC_SEM_BLOCK_STATEMENT_H_
 
 #include <vector>
+#include <cstddef>
 
 #include "src/sem/statement.h"
 
