diff --git a/src/wayland-eglsurface.c b/src/wayland-eglsurface.c
index 01c9cb3..21c5308 100644
--- a/src/wayland-eglsurface.c
+++ b/src/wayland-eglsurface.c
@@ -40,6 +40,10 @@
 
 #define WL_EGL_WINDOW_DESTROY_CALLBACK_SINCE 3
 
+#ifndef EGL_WAYLAND_Y_INVERTED_WL
+#define EGL_WAYLAND_Y_INVERTED_WL		0x31DB /* eglQueryWaylandBufferWL attribute */
+#endif
+
 struct wl_list wlEglSurfaceList = WL_LIST_INIT(&wlEglSurfaceList);
 
 EGLBoolean wlEglIsWlEglSurface(WlEglSurface *surface)
