--- chromium-70.0.3528.4/third_party/crc32c/src/src/crc32c_arm64_check.h.aarch64	2018-08-21 11:51:17.000000000 +0200
+++ chromium-70.0.3528.4/third_party/crc32c/src/src/crc32c_arm64_check.h	2018-08-27 03:33:34.550997403 +0200
@@ -33,7 +33,15 @@
   // From 'arch/arm64/include/uapi/asm/hwcap.h' in Linux kernel source code.
   constexpr unsigned long kHWCAP_PMULL = 1 << 4;
   constexpr unsigned long kHWCAP_CRC32 = 1 << 7;
-  unsigned long hwcap = (&getauxval != nullptr) ? getauxval(AT_HWCAP) : 0;
+  unsigned long hwcap =
+#if !HAVE_STRONG_GETAUXVAL
+    (&getauxval != nullptr) ?
+#endif
+    getauxval(AT_HWCAP)
+#if !HAVE_STRONG_GETAUXVAL
+    : 0
+#endif
+    ;
   return (hwcap & (kHWCAP_PMULL | kHWCAP_CRC32)) ==
          (kHWCAP_PMULL | kHWCAP_CRC32);
 #else
