diff --git a/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.cc b/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.cc
index 2a7b125..9deeb34 100644
--- a/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.cc
+++ b/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.cc
@@ -40,8 +40,8 @@
 #include "rtc_base/time_utils.h"
 #include "rtc_base/trace_event.h"
 #include "third_party/libyuv/include/libyuv/convert.h"
-#include "vpx/vp8cx.h"
-#include "vpx/vpx_encoder.h"
+#include <vpx/vp8cx.h>
+#include <vpx/vpx_encoder.h>
 
 namespace webrtc {
 
diff --git a/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.h b/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.h
index 954c044..50f1ab0 100644
--- a/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.h
+++ b/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.h
@@ -29,7 +29,7 @@
 #include "modules/video_coding/svc/scalable_video_controller.h"
 #include "modules/video_coding/utility/framerate_controller.h"
 #include "rtc_base/experiments/encoder_info_settings.h"
-#include "vpx/vp8cx.h"
+#include <vpx/vp8cx.h>
 
 namespace webrtc {
 
diff --git a/third_party/webrtc/modules/video_coding/codecs/vp9/vp9.cc b/third_party/webrtc/modules/video_coding/codecs/vp9/vp9.cc
index d9caf0f..e8557da 100644
--- a/third_party/webrtc/modules/video_coding/codecs/vp9/vp9.cc
+++ b/third_party/webrtc/modules/video_coding/codecs/vp9/vp9.cc
@@ -18,9 +18,9 @@
 #include "modules/video_coding/codecs/vp9/libvpx_vp9_decoder.h"
 #include "modules/video_coding/codecs/vp9/libvpx_vp9_encoder.h"
 #include "rtc_base/checks.h"
-#include "vpx/vp8cx.h"
-#include "vpx/vp8dx.h"
-#include "vpx/vpx_codec.h"
+#include <vpx/vp8cx.h>
+#include <vpx/vp8dx.h>
+#include <vpx/vpx_codec.h>
 
 namespace webrtc {
 
diff --git a/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.h b/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.h
index 954c044..50f1ab0 100644
--- a/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.h
+++ b/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.h
@@ -29,7 +29,7 @@
 #include "modules/video_coding/svc/scalable_video_controller.h"
 #include "modules/video_coding/utility/framerate_controller.h"
 #include "rtc_base/experiments/encoder_info_settings.h"
-#include "vpx/vp8cx.h"
+#include <vpx/vp8cx.h>
 
 namespace webrtc {
 
diff --git a/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_decoder.h b/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_decoder.h
index f26f427..4a5da0e 100644
--- a/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_decoder.h
+++ b/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_decoder.h
@@ -19,7 +19,7 @@
 #include "common_video/include/video_frame_buffer_pool.h"
 #include "modules/video_coding/codecs/vp9/include/vp9.h"
 #include "modules/video_coding/codecs/vp9/vp9_frame_buffer_pool.h"
-#include "vpx/vp8cx.h"
+#include <vpx/vp8cx.h>
 
 namespace webrtc {
 
diff --git a/third_party/webrtc/modules/video_coding/codecs/vp8/default_temporal_layers_unittest.cc b/third_party/webrtc/modules/video_coding/codecs/vp8/default_temporal_layers_unittest.cc
index a18ac40..f5978c3 100644
--- a/third_party/webrtc/modules/video_coding/codecs/vp8/default_temporal_layers_unittest.cc
+++ b/third_party/webrtc/modules/video_coding/codecs/vp8/default_temporal_layers_unittest.cc
@@ -22,7 +22,7 @@
 #include "test/field_trial.h"
 #include "test/gmock.h"
 #include "test/gtest.h"
-#include "vpx/vp8cx.h"
+#include <vpx/vp8cx.h>
 
 // TODO(bugs.webrtc.org/10582): Test the behavior of UpdateConfiguration().
 
diff --git a/third_party/webrtc/modules/video_coding/codecs/vp8/screenshare_layers_unittest.cc b/third_party/webrtc/modules/video_coding/codecs/vp8/screenshare_layers_unittest.cc
index 88ef9b8..a6f27ee 100644
--- a/third_party/webrtc/modules/video_coding/codecs/vp8/screenshare_layers_unittest.cc
+++ b/third_party/webrtc/modules/video_coding/codecs/vp8/screenshare_layers_unittest.cc
@@ -26,7 +26,7 @@
 #include "system_wrappers/include/metrics.h"
 #include "test/gmock.h"
 #include "test/gtest.h"
-#include "vpx/vp8cx.h"
+#include <vpx/vp8cx.h>
 
 using ::testing::_;
 using ::testing::ElementsAre;
diff --git a/third_party/webrtc/modules/video_coding/codecs/vp8/libvpx_vp8_encoder.h b/third_party/webrtc/modules/video_coding/codecs/vp8/libvpx_vp8_encoder.h
index ed80eac..df8b325 100644
--- a/third_party/webrtc/modules/video_coding/codecs/vp8/libvpx_vp8_encoder.h
+++ b/third_party/webrtc/modules/video_coding/codecs/vp8/libvpx_vp8_encoder.h
@@ -28,8 +28,8 @@
 #include "rtc_base/experiments/cpu_speed_experiment.h"
 #include "rtc_base/experiments/encoder_info_settings.h"
 #include "rtc_base/experiments/rate_control_settings.h"
-#include "vpx/vp8cx.h"
-#include "vpx/vpx_encoder.h"
+#include <vpx/vp8cx.h>
+#include <vpx/vpx_encoder.h>
 
 namespace webrtc {
 
diff --git a/third_party/webrtc/modules/video_coding/codecs/vp8/libvpx_vp8_encoder.cc b/third_party/webrtc/modules/video_coding/codecs/vp8/libvpx_vp8_encoder.cc
index 6a8a901..5c7db58 100644
--- a/third_party/webrtc/modules/video_coding/codecs/vp8/libvpx_vp8_encoder.cc
+++ b/third_party/webrtc/modules/video_coding/codecs/vp8/libvpx_vp8_encoder.cc
@@ -40,7 +40,7 @@
 #include "rtc_base/trace_event.h"
 #include "system_wrappers/include/field_trial.h"
 #include "third_party/libyuv/include/libyuv/scale.h"
-#include "vpx/vp8cx.h"
+#include <vpx/vp8cx.h>
 
 namespace webrtc {
 namespace {
diff --git a/third_party/webrtc/modules/video_coding/codecs/interface/libvpx_interface.h b/third_party/webrtc/modules/video_coding/codecs/interface/libvpx_interface.h
index 3dea24d..903c28d 100644
--- a/third_party/webrtc/modules/video_coding/codecs/interface/libvpx_interface.h
+++ b/third_party/webrtc/modules/video_coding/codecs/interface/libvpx_interface.h
@@ -15,10 +15,10 @@
 
 #include <memory>
 
-#include "vpx/vp8cx.h"
-#include "vpx/vpx_codec.h"
-#include "vpx/vpx_encoder.h"
-#include "vpx/vpx_image.h"
+#include <vpx/vp8cx.h>
+#include <vpx/vpx_codec.h>
+#include <vpx/vpx_encoder.h>
+#include <vpx/vpx_image.h>
 
 namespace webrtc {
 
diff --git a/media/video/vpx_video_encoder.cc b/media/video/vpx_video_encoder.cc
index f3b8b35..e4a7bf6 100644
--- a/media/video/vpx_video_encoder.cc
+++ b/media/video/vpx_video_encoder.cc
@@ -14,7 +14,7 @@
 #include "media/base/bind_to_current_loop.h"
 #include "media/base/video_frame.h"
 #include "media/base/video_util.h"
-#include "third_party/libvpx/source/libvpx/vpx/vp8cx.h"
+#include <vpx/vp8cx.h>
 #include "third_party/libyuv/include/libyuv/convert.h"
 
 namespace media {
diff --git a/media/cast/sender/vp8_encoder.cc b/media/cast/sender/vp8_encoder.cc
index 2131145..968cda1 100644
--- a/media/cast/sender/vp8_encoder.cc
+++ b/media/cast/sender/vp8_encoder.cc
@@ -7,7 +7,7 @@
 #include "base/logging.h"
 #include "media/base/video_frame.h"
 #include "media/cast/constants.h"
-#include "third_party/libvpx/source/libvpx/vpx/vp8cx.h"
+#include <vpx/vp8cx.h>
 
 namespace media {
 namespace cast {
diff --git a/content/renderer/pepper/video_encoder_shim.cc b/content/renderer/pepper/video_encoder_shim.cc
index 185481b..abe8688 100644
--- a/content/renderer/pepper/video_encoder_shim.cc
+++ b/content/renderer/pepper/video_encoder_shim.cc
@@ -17,8 +17,8 @@
 #include "base/threading/thread_task_runner_handle.h"
 #include "content/renderer/pepper/pepper_video_encoder_host.h"
 #include "content/renderer/render_thread_impl.h"
-#include "third_party/libvpx/source/libvpx/vpx/vp8cx.h"
-#include "third_party/libvpx/source/libvpx/vpx/vpx_encoder.h"
+#include <vpx/vp8cx.h>
+#include <vpx/vpx_encoder.h>
 #include "ui/gfx/geometry/size.h"
 
 namespace content {
diff --git a/remoting/codec/webrtc_video_encoder_vpx.cc b/remoting/codec/webrtc_video_encoder_vpx.cc
index d77b908..d4dedfd 100644
--- a/remoting/codec/webrtc_video_encoder_vpx.cc
+++ b/remoting/codec/webrtc_video_encoder_vpx.cc
@@ -16,8 +16,8 @@
 #include "build/chromeos_buildflags.h"
 #include "remoting/base/util.h"
 #include "remoting/proto/video.pb.h"
-#include "third_party/libvpx/source/libvpx/vpx/vp8cx.h"
-#include "third_party/libvpx/source/libvpx/vpx/vpx_encoder.h"
+#include <vpx/vp8cx.h>
+#include <vpx/vpx_encoder.h>
 #include "third_party/libyuv/include/libyuv/convert_from_argb.h"
 #include "third_party/webrtc/modules/desktop_capture/desktop_frame.h"
 #include "third_party/webrtc/modules/desktop_capture/desktop_geometry.h"
diff --git a/remoting/codec/video_encoder_vpx.cc b/remoting/codec/video_encoder_vpx.cc
index 30b5f97..7827461 100644
--- a/remoting/codec/video_encoder_vpx.cc
+++ b/remoting/codec/video_encoder_vpx.cc
@@ -12,8 +12,8 @@
 #include "base/system/sys_info.h"
 #include "remoting/base/util.h"
 #include "remoting/proto/video.pb.h"
-#include "third_party/libvpx/source/libvpx/vpx/vp8cx.h"
-#include "third_party/libvpx/source/libvpx/vpx/vpx_encoder.h"
+#include <vpx/vp8cx.h>
+#include <vpx/vpx_encoder.h>
 #include "third_party/libyuv/include/libyuv/convert_from_argb.h"
 #include "third_party/webrtc/modules/desktop_capture/desktop_frame.h"
 #include "third_party/webrtc/modules/desktop_capture/desktop_geometry.h"
diff --git a/third_party/blink/renderer/modules/mediarecorder/vpx_encoder.h b/third_party/blink/renderer/modules/mediarecorder/vpx_encoder.h
index b544dbf..77233eb 100644
--- a/third_party/blink/renderer/modules/mediarecorder/vpx_encoder.h
+++ b/third_party/blink/renderer/modules/mediarecorder/vpx_encoder.h
@@ -8,8 +8,8 @@
 #include "base/sequenced_task_runner.h"
 #include "third_party/blink/renderer/modules/mediarecorder/video_track_recorder.h"
 #include "third_party/blink/renderer/platform/wtf/vector.h"
-#include "third_party/libvpx/source/libvpx/vpx/vp8cx.h"
-#include "third_party/libvpx/source/libvpx/vpx/vpx_encoder.h"
+#include <vpx/vp8cx.h>
+#include <vpx/vpx_encoder.h>
 
 namespace blink {
 
--
