From caa69c39d808b56f85ddb24679a6d50c6a32ccf0 Mon Sep 17 00:00:00 2001
From: Qball Cow <qball@blame.services>
Date: Tue, 7 Oct 2025 23:04:35 +0200
Subject: [PATCH] [Script] fix missing and wrong free

---
 source/modes/script.c | 33 +++++++++++++++++----------------
 1 file changed, 17 insertions(+), 16 deletions(-)

diff --git a/source/modes/script.c b/source/modes/script.c
index 14d28b3e1..c6d9699d7 100644
--- a/source/modes/script.c
+++ b/source/modes/script.c
@@ -26,6 +26,7 @@
  */
 
 /** The log domain of this dialog. */
+#include "glib.h"
 #define G_LOG_DOMAIN "Modes.Script"
 
 #include "modes/script.h"
@@ -321,6 +321,19 @@ static void script_mode_reset_highlight(Mode *sw) {
   rmpd->active_list = NULL;
 }
 
+static void script_mode_free_entry_list (DmenuScriptEntry *list, unsigned int length )
+{
+    for ( unsigned int i = 0; i < length; i++){
+      g_free(list[i].entry);
+      g_strfreev(list[i].icon_name);
+      g_free(list[i].display);
+      g_free(list[i].meta);
+      g_free(list[i].info);
+
+    }
+    g_free(list);
+
+}
 static ModeMode script_mode_result(Mode *sw, int mretv, char **input,
                                    unsigned int selected_line) {
   ScriptModePrivateData *rmpd = (ScriptModePrivateData *)sw->private_data;
@@ -387,15 +387,7 @@ static ModeMode script_mode_result(Mode
 
   // If a new list was generated, use that an loop around.
   if (new_list != NULL) {
-    for (unsigned int i = 0; i < rmpd->cmd_list_length; i++) {
-      g_free(rmpd->cmd_list[i].entry);
-      g_free(rmpd->cmd_list[i].icon_name);
-      g_free(rmpd->cmd_list[i].display);
-      g_free(rmpd->cmd_list[i].meta);
-      g_free(rmpd->cmd_list[i].info);
-    }
-    g_free(rmpd->cmd_list);
-
+    script_mode_free_entry_list(rmpd->cmd_list, rmpd->cmd_list_length);
     rmpd->cmd_list = new_list;
     rmpd->cmd_list_length = new_length;
     if (keep_selection) {
@@ -396,6 +396,7 @@ static ModeMode script_mode_result(Mode
       rofi_view_set_selected_line(rofi_view_get_active(), 0);
     }
     if (keep_filter == FALSE) {
+      script_mode_free_entry_list(new_list, new_length);
       g_free(*input);
       *input = NULL;
     }
@@ -407,13 +408,7 @@ static ModeMode script_mode_result(Mode
 static void script_mode_destroy(Mode *sw) {
   ScriptModePrivateData *rmpd = (ScriptModePrivateData *)sw->private_data;
   if (rmpd != NULL) {
-    for (unsigned int i = 0; i < rmpd->cmd_list_length; i++) {
-      g_free(rmpd->cmd_list[i].entry);
-      g_free(rmpd->cmd_list[i].icon_name);
-      g_free(rmpd->cmd_list[i].display);
-      g_free(rmpd->cmd_list[i].meta);
-    }
-    g_free(rmpd->cmd_list);
+    script_mode_free_entry_list(rmpd->cmd_list, rmpd->cmd_list_length);
     g_free(rmpd->message);
     g_free(rmpd->prompt);
     g_free(rmpd->data);
