From 0445d97c8ad66c9cbb2c6f88c52aadc0b01a1f15 Mon Sep 17 00:00:00 2001
From: Naveen Saini <naveen.kumar.saini@intel.com>
Date: Tue, 11 Oct 2022 16:43:22 +0800
Subject: [PATCH] CMakeLists.txt: link with libclang-cpp library instead

Upstream-Status: Inappropriate

Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f6199704..f979f1c1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -377,7 +377,7 @@ if (${LLVM_VERSION_NUMBER} VERSION_LESS
     message(FATAL_ERROR "ISPC requires LLVM 16.0.0 or later")
 endif()
 
-set(CLANG_LIBRARY_LIST clangFrontend clangBasic clangEdit clangLex clangSupport clangASTMatchers)
+set(CLANG_LIBRARY_LIST clang-cpp)
 
 set(LLVM_COMPONENTS engine ipo bitreader bitwriter instrumentation linker option frontendopenmp passes)
 
-- 
2.37.3
