diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1319479..6b082db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -262,7 +262,7 @@ get_llvm_libfiles(LLVM_LIBRARY_LIST ${LLVM_COMPONENTS})
 get_llvm_cppflags(LLVM_CPP_FLAGS)
 
 generate_target_builtins(BUILTIN_FILES ${ISPC_TARGETS})
-generate_common_builtins(BUILTIN_CPP_FILES)
+#generate_common_builtins(BUILTIN_CPP_FILES)
 generate_stdlib(STDLIB_FILES ${ISPC_MASKS})
 
 add_executable(${PROJECT_NAME} ${BUILTIN_FILES} ${BUILTIN_CPP_FILES}
