diff --git a/./third_party/webrtc/modules/desktop_capture/desktop_capture_options.cc b/./third_party/webrtc/modules/desktop_capture/desktop_capture_options.cc
index 8a33807..41181e8 100644
--- a/./third_party/webrtc/modules/desktop_capture/desktop_capture_options.cc
+++ b/./third_party/webrtc/modules/desktop_capture/desktop_capture_options.cc
@@ -27,9 +27,6 @@ DesktopCaptureOptions& DesktopCaptureOptions::operator=(
 // static
 DesktopCaptureOptions DesktopCaptureOptions::CreateDefault() {
   DesktopCaptureOptions result;
-#if defined(USE_X11)
-  result.set_x_display(SharedXDisplay::CreateDefault());
-#endif
 #if defined(WEBRTC_MAC) && !defined(WEBRTC_IOS)
   result.set_configuration_monitor(new DesktopConfigurationMonitor());
   result.set_full_screen_chrome_window_detector(
