diff --git a/src/libslic3r/AppConfig.cpp b/src/libslic3r/AppConfig.cpp
index 18dc7396e..95483e844 100644
--- a/src/libslic3r/AppConfig.cpp
+++ b/src/libslic3r/AppConfig.cpp
@@ -1,5 +1,6 @@
 #include "libslic3r/libslic3r.h"
 #include "libslic3r/Utils.hpp"
+#include "libslic3r/I18N.hpp"
 #include "AppConfig.hpp"
 #include "Exception.hpp"
 #include "LocalesUtils.hpp"
diff --git a/src/slic3r/GUI/CreateMMUTiledCanvas.cpp b/src/slic3r/GUI/CreateMMUTiledCanvas.cpp
index be157eafa..d63eb110e 100644
--- a/src/slic3r/GUI/CreateMMUTiledCanvas.cpp
+++ b/src/slic3r/GUI/CreateMMUTiledCanvas.cpp
@@ -40,6 +40,7 @@
 #include <wx/odcombo.h>
 #include <wx/textctrl.h>
 #include <wx/wrapsizer.h>
+#include <wx/rawbmp.h>
 #include "wxExtensions.hpp"
 
 #include <boost/filesystem/path.hpp>
--- a/src/libslic3r/ExtrusionEntity.hpp
+++ b/src/libslic3r/ExtrusionEntity.hpp
@@ -4,6 +4,7 @@
 #include "libslic3r.h"
 #include "Polygon.hpp"
 #include "Polyline.hpp"
+#include "PrintConfig.hpp"
 
 #include <assert.h>
 #include <string_view>
